### PR TITLE
CIV-9039 generate judgment by determination

### PR DESCRIPTION
**Pull request by paul-pearson**
### JIRA link ###

https://tools.hmcts.net/jira/browse/CIV-9039

### Change description ###

Generate judgment by determination

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts [hunk 1]
```diff
@@ -3,22 +3,72 @@ import { Claim } from 'common/models/claim';
...
+  extends ClaimantResponsePaymentPlanDetails, CCDClaim {
+  ccjPaymentPaidSomeOption?: string;
+  ccjPaymentPaidSomeAmount?: string;
+  ccjJudgmentAmountClaimFee?: string;
+  ccjJudgmentLipInterest?: string;
+}
+
+export const translateClaimantResponseRequestJudgementByAdmissionOrDeterminationToCCD = (claim: Claim, claimFee: number): ClaimantResponseRequestJudgementByAdmissionOrDeterminationToCCD => {
+  let paymentPlanDetails: ClaimantResponsePaymentPlanDetails;
+  if (claim.claimantResponse?.chooseHowToProceed?.option === ChooseHowProceed.REQUEST_A_CCJ) {
```
Comment (jeswanth-hmcts) [inline src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts:38]: create a claimantresponse instance and use the existing methods  isCCJRequested to  check  the above condition instead doing it seprately
const claimantResponse = Object.assign(new ClaimantResponse(), claim.claimantResponse);

File: src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts [hunk 2]
```diff
@@ -3,22 +3,72 @@ import { Claim } from 'common/models/claim';
...
+    let paymentIntention: PaymentIntention;
+    if (claim.claimantResponse.fullAdmitSetDateAcceptPayment?.option === YesNo.YES) {
+      paymentIntention = claim.getPaymentIntention();
+    } else if (claim.claimantResponse.courtProposedPlan?.decision === CourtProposedPlanOptions.ACCEPT_REPAYMENT_PLAN) {
```
Comment (jeswanth-hmcts) [inline src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts:42]: for this also

File: src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts [hunk 3]
```diff
@@ -3,22 +3,72 @@ import { Claim } from 'common/models/claim';
...
+      if (claim.claimantResponse.courtDecision == RepaymentDecisionType.IN_FAVOUR_OF_DEFENDANT) {
```
Comment (jeswanth-hmcts) [inline src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts:43]: for this also

File: src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts [hunk 4]
```diff
@@ -3,22 +3,72 @@ import { Claim } from 'common/models/claim';
...
+        paymentIntention = claim.getPaymentIntention();
+      } else if (claim.claimantResponse.courtDecision == RepaymentDecisionType.IN_FAVOUR_OF_CLAIMANT) {
```
Comment (jeswanth-hmcts) [inline src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts:45]: for this also , if not exists create a method inside the class

File: src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts (hunk 1 above)
Comment (paul-pearson) [reply src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts]: Fixed as suggested.

File: src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts (hunk 2 above)
Comment (paul-pearson) [reply src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts]: As above.

File: src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts (hunk 3 above)
Comment (paul-pearson) [reply src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts]: As above.

File: src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts (hunk 4 above)
Comment (paul-pearson) [reply src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts]: As above.

File: src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts [hunk 5]
```diff
@@ -3,22 +3,74 @@ import { Claim } from 'common/models/claim';
...
+  ccjPaymentPaidSomeOption?: string;
+  ccjPaymentPaidSomeAmount?: string;
+  ccjJudgmentAmountClaimFee?: string;
+  ccjJudgmentLipInterest?: string;
+}
+
+export const translateClaimantResponseRequestJudgementByAdmissionOrDeterminationToCCD = (claim: Claim, claimFee: number): ClaimantResponseRequestJudgementByAdmissionOrDeterminationToCCD => {
+  let paymentPlanDetails: ClaimantResponsePaymentPlanDetails;
+  const claimantResponse = Object.assign(new ClaimantResponse(), claim.claimantResponse);
+  if (claimantResponse.chooseHowToProceed?.option === ChooseHowProceed.REQUEST_A_CCJ) {
```
Comment (jeswanth-hmcts) [inline src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts:40]: The above condition be replaced by below method claimantResponse.isCCJRequested()

File: src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts [hunk 6]
```diff
@@ -3,22 +3,74 @@ import { Claim } from 'common/models/claim';
...
+    let paymentIntention: PaymentIntention;
+    if (claimantResponse.fullAdmitSetDateAcceptPayment?.option === YesNo.YES) {
+      paymentIntention = claim.getPaymentIntention();
+    } else if (claimantResponse.courtProposedPlan?.decision === CourtProposedPlanOptions.ACCEPT_REPAYMENT_PLAN) {
+      if (claimantResponse.courtDecision == RepaymentDecisionType.IN_FAVOUR_OF_DEFENDANT) {
```
Comment (jeswanth-hmcts) [inline src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts:45]: this condition can be replaced with this method isCourtDecisionInFavourOfDefendant

File: src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts [hunk 7]
```diff
@@ -3,22 +3,74 @@ import { Claim } from 'common/models/claim';
...
+        paymentIntention = claim.getPaymentIntention();
+      } else if (claimantResponse.courtDecision == RepaymentDecisionType.IN_FAVOUR_OF_CLAIMANT) {
```
Comment (jeswanth-hmcts) [inline src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts:47]: replaced by this method isCourtDecisionInFavourOfClaimant

File: src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts [hunk 8]
```diff
@@ -3,22 +3,74 @@ import { Claim } from 'common/models/claim';
...
+}
+
+export const translateClaimantResponseRequestJudgementByAdmissionOrDeterminationToCCD = (claim: Claim, claimFee: number): ClaimantResponseRequestJudgementByAdmissionOrDeterminationToCCD => {
+  let paymentPlanDetails: ClaimantResponsePaymentPlanDetails;
+  const claimantResponse = Object.assign(new ClaimantResponse(), claim.claimantResponse);
+  if (claimantResponse.chooseHowToProceed?.option === ChooseHowProceed.REQUEST_A_CCJ) {
+    let paymentIntention: PaymentIntention;
+    if (claimantResponse.fullAdmitSetDateAcceptPayment?.option === YesNo.YES) {
+      paymentIntention = claim.getPaymentIntention();
+    } else if (claimantResponse.courtProposedPlan?.decision === CourtProposedPlanOptions.ACCEPT_REPAYMENT_PLAN) {
```
Comment (jeswanth-hmcts) [inline src/main/services/translation/claimantResponse/ccdRequestJudgementTranslation.ts:44]: may be you can use this smethod isClaimantAcceptsCourtDecision

**Review by jeswanth-hmcts — DISMISSED** [review]

**Review by jeswanth-hmcts — APPROVED** [review]

**Review by jeswanth-hmcts — APPROVED** [review]

**Pull request closed without merging**